pallets/gas/src/migrations.rs

```diff
@@ -0,0 +1,680 @@
+                    waitlist_storage_prefix.to_vec(),
+                    |mut key| Ok(WaitlistKey::decode(&mut key)?.into()),
+                )
+                .collect::<BTreeSet<MessageId>>();
+
+                let mailbox_storage_prefix = storage_prefix(
+                    PALLET_GEAR_MESSENGER_NAME.as_bytes(),
+                    MAILBOX_STORAGE_NAME.as_bytes(),
+                );
+                let mailbox_keys = KeyPrefixIterator::new(
+                    mailbox_storage_prefix.to_vec(),
+                    mailbox_storage_prefix.to_vec(),
+                    |mut key| Ok(WaitlistKey::decode(&mut key)?.into()),
+                )
+                .collect::<BTreeSet<MessageId>>();
+
+                let dispatch_stash_storage_prefix = storage_prefix(
+                    PALLET_GEAR_MESSENGER_NAME.as_bytes(),
+                    DISPATCH_STASH_STORAGE_NAME.as_bytes(),
+                );
+                let dispatch_stash_keys = KeyPrefixIterator::new(
+                    dispatch_stash_storage_prefix.to_vec(),
+                    dispatch_stash_storage_prefix.to_vec(),
+                    |mut key| DispatchStashKey::decode(&mut key),
+                )
+                .collect::<BTreeSet<_>>();
+
+                db_reads = db_reads
+                    .saturating_add(waitlist_keys.len() as u64)
+                    .saturating_add(mailbox_keys.len() as u64)
+                    .saturating_add(dispatch_stash_keys.len() as u64);
+
+                // A function that tries to derive the actual lock type throuhg storages scan
+                // Caveat: for simplicity sake we do not scan programs reservation maps:
+                // - assuming a lock to be a Reservation if the key is not found in other storages.
+                let appropriate_lock_id = |node_id: &NodeId| -> LockId {
+                    match node_id {
+                        NodeId::Node(msg_id) => {
+                            if mailbox_keys.contains(msg_id) {
+                                LockId::Mailbox
+                            } else if waitlist_keys.contains(msg_id) {
+                                LockId::Waitlist
+                            } else if dispatch_stash_keys.contains(msg_id) {
+                                LockId::DispatchStash
+                            } else {
+                                // Likely unreachable branch
+                                LockId::Reservation
+                            }
+                        }
+                        _ => LockId::Reservation,
+                    }
+                };
+
+                let mut translated = 0_u64;
+                GasNodes::<T>::translate::<OldGasNode<AccountIdOf<T>, NodeId, Balance>, _>(
+                    |key, old_value| {
+                        let mut new_lock = NodeLock::<Balance>::zero();
+                        let new_value = match old_value {
+                            OldGasNode::External {
```

Comment:
```suggestion
                            v1::GasNode::External {
```
Seems more readable for me taking in account that there will be more migrations in the file soon

Reply:
> Seems more readable for me taking in account that there will be more migrations in the file soon

Yeah, but the problem is that we are importing the post-migration `GasNode` from `gear-common` into the namespace, as well - in any case we'd need to use different names for these two types.